uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Page/AbsenceType.php

```diff
@@ -16,6 +16,15 @@ class CRM_HRLeaveAndAbsences_Page_AbsenceType extends CRM_Core_Page_Basic {
   public function run() {
     CRM_Utils_System::setTitle(ts('Leave/Absence Types'));
     $this->absenceTypeService = new AbsenceTypeService();
+    $action = CRM_Utils_Request::retrieve('action', 'String', $this, FALSE, 'browse');
+    $this->assign('action', $action);
+
+    if ($action & CRM_Core_Action::BROWSE) {
```

Comment:
Why do we need to manually assign this now? seems it was done automatically before?

Reply:
This same page is used for both listing and modifying of records. As stated in the description, this allow us to render and manage the flow of the form. So we need to know what action was requested by user and render appropriate page.

Reply:
I understand, but before this change, the template has logic like `{if $action eq 1 or $action eq 2 or $action eq 8}`, which means the action variable was being sent to the template, why do we need to do this manually now?

Reply:
The existing implementation does not seem work and not so visible because we have a separate  template handling editing and updating of absence types. So we need to call the `assign` method for the variable to be available in the template.